main/pipeline/pipeline.h

```diff
@@ -31,8 +31,11 @@ std::vector<ast::ParsedFile> incrementalResolve(core::GlobalState &gs, std::vect
 std::vector<ast::ParsedFile> name(core::GlobalState &gs, std::vector<ast::ParsedFile> what,
                                   const options::Options &opts, bool skipConfigatron = false);
 
+// Note: `epoch` parameter is only used in LSP mode when preemptible is true. Do not use it elsewhere.
+// TODO(jvilk): Make this interface cleaner.
 ast::ParsedFilesOrCancelled typecheck(std::unique_ptr<core::GlobalState> &gs, std::vector<ast::ParsedFile> what,
-                                      const options::Options &opts, WorkerPool &workers);
+                                      const options::Options &opts, WorkerPool &workers, bool cancelable = false,
```

Comment:
`cancelable` is now an explicit parameter because the fast path calls `typecheck` but is not cancelable.